### PR TITLE
Fix attaching partition to a distributed table in schema

### LINKED ISSUE #1772: Cannot attach partition in a schema
`ALTER TABLE .. ATTACH PARTITION` errors out for tables in a schema.`

```
CREATE SCHEMA test;
CREATE TABLE test.part_test (col1 serial, col2 text, col3 timestamptz NOT NULL DEFAULT now()) PARTITION BY RANGE (col3);
SELECT create_distributed_table('test.part_test', 'col1');
CREATE TABLE test.part_test_p2017_11_08_0900 (col1 serial, col2 text, col3 timestamptz NOT NULL DEFAULT now());
ALTER TABLE test.part_test ATTACH PARTITION test.part_test_p2017_11_08_0900 FOR VALUES FROM ('2017-11-08 09:00:00+00') TO ('2017-11-08 10:00:00+00');
ERROR:  relation "part_test" does not exist
```

but works for tables in the public schema:
```
CREATE TABLE part_test (col1 serial, col2 text, col3 timestamptz NOT NULL DEFAULT now()) PARTITION BY RANGE (col3);
SELECT create_distributed_table('part_test', 'col1');
CREATE TABLE part_test_p2017_11_08_0900 (col1 serial, col2 text, col3 timestamptz NOT NULL DEFAULT now());
ALTER TABLE part_test ATTACH PARTITION part_test_p2017_11_08_0900 FOR VALUES FROM ('2017-11-08 09:00:00+00') TO ('2017-11-08 10:00:00+00');
```

(automatically distributes `part_test_p2017_11_08_0900`)

### PR DESCRIPTION
**Pull request by byucesoy**
Fixes #1772 

While attaching a partition to a distributed table in schema, we mistakenly
used unqualified name to find partitioned table's oid. This caused problems
while using partitioned tables with schemas. We are fixing this issue in
this PR.

### CODE REVIEW

File: src/test/regress/expected/multi_partitioning.out [hunk 1]
```diff
@@ -1255,3 +1255,86 @@ SELECT create_distributed_table('partitioning_test', 'id');
...
+-- make sure we can attach partitions to a distributed table in a schema
+CREATE SCHEMA partitioning_schema;
+CREATE TABLE partitioning_schema.schema_test(id int, time date) PARTITION BY RANGE (time);
+SELECT create_distributed_table('partitioning_schema.schema_test', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+CREATE TABLE partitioning_schema.schema_test_2009(id int, time date);
```
Comment (marcocitus) [inline src/test/regress/expected/multi_partitioning.out]: Maybe add some character that requires quotes, e.g. "schema-test-2009", to make sure quoting is working correctly.

File: src/backend/distributed/executor/multi_utility.c [hunk 2]
```diff
@@ -952,7 +952,7 @@ ProcessAlterTableStmtAttachPartition(AlterTableStmt *alterTableStatement)
 			{
 				Var *distributionColumn = DistPartitionKey(relationId);
 				char distributionMethod = DISTRIBUTE_BY_HASH;
-				char *relationName = get_rel_name(relationId);
+				char *relationName = generate_qualified_relation_name(relationId);
```
Comment (marcocitus) [inline src/backend/distributed/executor/multi_utility.c]: minor: variable should maybe also be `parentRelationName`

File: src/test/regress/expected/multi_partitioning.out [hunk 3]
```diff
@@ -1255,3 +1255,86 @@ SELECT create_distributed_table('partitioning_test', 'id');
...
+ALTER TABLE partitioning_schema.schema_test ATTACH PARTITION partitioning_schema.schema_test_2009 FOR VALUES FROM ('2009-01-01') TO ('2010-01-01');
```
Comment (onderkalaci) [inline src/test/regress/expected/multi_partitioning.out]: Would be nice to have one tests for each with `set search_path = ...` and attach partitions

**Comment by marcocitus** [comment]
This is now my favourite thing in 7.1 :)

**Pull request merged**